Formula/q/qrq.rb

```diff
@@ -0,0 +1,15 @@
+class Qrq < Formula
+  desc "High speed morse telegraphy trainer"
+  homepage "https://fkurz.net/ham/qrq.html"
+  url "https://fkurz.net/ham/qrq/qrq-0.3.5.tar.gz"
+  sha256 "3b068c960d06b254dca5a48c7813f1653fc53cb6f5ced641e007091a7ec08d4f"
+  license "GPL-2.0-or-later"
+
+  def install
+    system "make", "DESTDIR=#{prefix}", "USE_CA=YES", "USE_PA=NO", "OSX_PLATFORM=YES", "OSX_BUNDLE=NO"
+    system "make", "install", "DESTDIR=#{prefix}", "USE_CA=YES", "USE_PA=NO", "OSX_PLATFORM=YES", "OSX_BUNDLE=NO"
+  end
+  test do
+    system bin/"qrq", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I'm trying to figure out what might work as a test case here. The software interacts with the user through a curses interface and does not have any switches that can be invoked at startup.
According to their man-page they will try check for a configuration file and if it doesn't exist at startup, they'll create ~/.qrq/ . I don't know how that could be evaluate through the brew test mechanism.

Additionally, I'll need to learn how to differentiate building on Linux or OS X, to set the make switches accordingly. :)